### DISCUSSION TITLE
PHP program to get amount of requests sent to a users site to prevent a DDoS

### DISCUSSION THREAD

**Discussion by emullallphotography** [community/community#69735] [Programming Help]
```php
<?php 
 $requestSent = 0;
 while(true){
    $calc = $requestsSent*60;
    if($calc > 100){
     //Block request
   }else{
     //Allow Request
  }
 }
?>
```
### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by lennis-dev** [comment]
Hello @emullallphotography
you could for example write the IP address to a database or file for each request and if the number of requests is too high reject the request, if you need any code or help to implement it feel free to mention me.

With kind regards
Lennis

**Comment by ghost** [comment]
Hi,

I don't know much PHP I mainly focus on the front end instead of the backend, so if you could please help me I would  greatly appreciate it.

Where you the one who helped me with the website called PyTutorials?

**Reply by emullallphotography** [reply]
@lennis-dev,

is it possible to add you as a contributor?

**Reply by lennis-dev** [reply]
> Where you the one who helped me with the website called PyTutorials?

Yes, I think so

> is it possible to add you as a contributor?

Yes, is no problem

Kind regards

**Reply by ghost** [reply]
I added you 

https://github.com/Websitehosterio/websitehosterio.github.io/blob/main/request.php

**Reply by lennis-dev** [reply]
Okay, I have received and accepted your invitation, do you have a SQL server and do you want to host your website?

**Reply by ghost** [reply]
Only PHP, CSS and JavaScript. No SQL server. I am in school right now so I may reply late.

School for me ends around 2:30pm

**Reply by ghost** [reply]
Also I want to host my website

**Reply by lennis-dev** [reply]
I have implmented a code with the code I hope it helps you and the answer was helpful, if you have any questions feel free to ask me.

https://github.com/Websitehosterio/websitehosterio.github.io/commit/9659aa8b3234947776d84a76987df9877a2af289

Kind regards and happy Hacktoberfest

**Reply by emullallphotography** [reply]
Where did you learn PHP? Because I can’t find any place to learn it

**Reply by lennis-dev** [reply]
I read a lot of documentation. Other helpful sites that have also helped me are: php.net, w3schools.com and stackoverflow.com.

**Comment by emullallphotography** [comment]
Hi @lennis-dev, 

sorry to bug you but would it be possible to help me with some of the PHP that requires expertise like sending requests to get the users GitHub repository?

**Reply by lennis-dev** [reply]
Yes, I can do that, but what does "sending requests to get the users GitHub repository" mean?

**Comment by ghost** [comment]
I was going to create a website that allows users to add server side support to there GitHub page and it was going to get the GitHub repository and add it onto my server

**Reply by lennis-dev** [reply]
You can use git clone by shell_exec. Best read this page: https://www.php.net/manual/en/function.shell-exec.php

**Reply by ghost** [reply]
Why do I need to use shell_exec?

**Reply by lennis-dev** [reply]
I thought you want to clone a well repo with PHP for this it is recommended to use shell_exec()

**Reply by ghost** [reply]
Yes, I did thanks I was just curious why you didn't use exec
